### ISSUE TITLE
SC2034 false positive

### ISSUE THREAD

**Issue by bje-** [koalaman/shellcheck#1019]
- Rule SC2034 (online)

### Here's a snippet that shows the problem:

```
#!/bin/bash

declare -A foo
foo[123]=456
arr=foo
var="$arr[123]"
echo "${!var}"
```

#### Here's what shellcheck currently says:

Line 4:
foo[123]=456
^-- SC2034: foo appears unused. Verify it or export it.

#### Here's what I wanted or expected to see:

Nothing. I have several associative arrays and want to choose between them at runtime.

**Comment by koalaman** [comment]
ShellCheck intentionally does not try to work out whether or not variables are used through indirection. See #1017 for more about this.

For scripts relying heavily on this, consider disabling the warning script wide.
